### PR TITLE
Custom Password Reset Token Generator

### PR DESCRIPTION
**Pull request by zanemcca**
I am using the custom token generator for email verification and I need to do the same for the password reset token. So I created this PR to address the issue.

I added a prototype function `requestPasswordReset` to user.
The format and options follow that of the `verify` prototype which allows for consistency when implementing the verify and reset features.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by zanemcca** [comment]
@altsang Could I get some feedback on this?

**Comment by zanemcca** [comment]
We require this functionality for our upcoming release so I would really appreciate some feedback as soon as possible so that we can help to alleviate any issues you may have with this addition.

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
> I am using the custom token generator for email verification and I need to do the same for the password reset token. So I created this PR to address the issue.

> I added a prototype function requestPasswordReset to user.
The format and options follow that of the verify prototype which allows for consistency when implementing the verify and reset features.

@zanemcca thank you for the pull request. 

@ritch could you please review? You are more familiar with this part of LoopBack than I am.

**Comment by zanemcca** [comment]
@ritch how do we go about fixing these errors?
The details links are resulting in a 404 and all of the tests passed for me on my local machine.

**Comment by bajtos** [comment]
Sorry for the long delay. I'm reassigning this patch to myself, will take a closer look.

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by bajtos** [comment]
@slnode test please

File: common/models/user.js [hunk 1]
```diff
@@ -438,6 +438,65 @@ module.exports = function(User) {
...
+
+    var user = this;
+    var userModel = this.constructor;
+    var ttl = userModel.settings.resetPasswordTokenTTL || DEFAULT_RESET_PW_TTL;
+
+    // Set the email to the user if it is not set
+    options.to = options.to || user.email;
+
+    // Set default token generator function if one is not provided
+    var tokenGenerator = options.generateVerificationToken || userModel.generateVerificationToken;
```
Comment (bajtos) [inline common/models/user.js]: So far, I believe we were not allowing user-provided `options` to provide functions to customize the behaviour. One of the reasons is that such functions cannot be specified via REST API.

Instead, we are expecting applications to subclass the built-in `User` model and change the behaviour by overriding certain functions, for example `User.prototype.createAccessToken`.

I would prefer to keep that approach, unless you have a specific need for providing different `tokenGenerator` functions on per-request basis.

Thoughts?

**Comment by bajtos** [comment]
I am closing this pull request as abandoned. Feel free to reopen if/when you get time to work on it again.

**Pull request closed without merging**